Index: lib/on-off-switch/index.coffee
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/on-off-switch/index.coffee	(revision dcc2774e5003aecfae162b6da17727cc6b5fb483)
+++ lib/on-off-switch/index.coffee	(revision )
@@ -17,32 +17,30 @@
         input: {}
 
     render: ->
-        inputId = if @props.input.id? then @props.input.id else _.uniqueId 'onoffswitch_'
         inputProps = _.extend @props.input,
             type: 'checkbox'
             name: 'onoffswitch'
-            id: inputId
             if @props.input.className?
                 className: "onoffswitch-checkbox #{@props.input.className}"
             else
                 className: 'onoffswitch-checkbox'
 
         r.div {className: 'onoffswitch'}, [
-            r.input inputProps,
             r.label
                 className: 'onoffswitch-label'
-                htmlFor: inputId
                 [
+                    r.input inputProps,
                     r.span
                         className: 'onoffswitch-inner'
                         [
                             r.span {className: 'onoffswitch-inner-on'},
                                 @props.i18n.t 'onoffswitch_label_on'
+                            r.span {className: 'onoffswitch-switch-outer'},
+                                r.span
+                                    className: 'onoffswitch-switch'
                             r.span {className: 'onoffswitch-inner-off'},
                                 @props.i18n.t 'onoffswitch_label_off'
                         ]
-                    r.span
-                        className: 'onoffswitch-switch'
                 ]
         ]
 
Index: lib/on-off-switch/styles/onoffswitch.styl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/on-off-switch/styles/onoffswitch.styl	(revision dcc2774e5003aecfae162b6da17727cc6b5fb483)
+++ lib/on-off-switch/styles/onoffswitch.styl	(revision )
@@ -1,18 +1,25 @@
 @require 'stylus-colors/styles/colors.styl'
 @require 'stylus-dimensions/styles/dimensions.styl'
 
+onoffswitch-transition-duration = 0.3s
+onoffswitch-element-height = 2.25em
+onoffswitch-switch-width = 0.6em
+onoffswitch-switch-outer-width = 2.2em // 2 * switch-width + 4 * inner-margin
+onoffswitch-switch-inner-height = 1.75em
+onoffswitch-switch-inner-margin = 0.25em
+onoffswitch-switch-unchecked-left-margin = 1.35em //outer-width - switch-width - inner-margin
+
 .onoffswitch
     position relative
-    min-width onoffswitch-element-width
-    -webkit-user-select none
+    user-select none
 
     .onoffswitch-checkbox
         display none
-        &:checked + .onoffswitch-label .onoffswitch-inner
+        &:checked ~ .onoffswitch-inner
             margin-left 0
-        &:checked + .onoffswitch-label .onoffswitch-switch
-            right 0px
+            & .onoffswitch-switch
+                margin-left onoffswitch-switch-inner-margin
-            
+
     .onoffswitch-label
         display block
         overflow hidden
@@ -28,7 +35,6 @@
     .onoffswitch-inner-on, .onoffswitch-inner-off
         display block
         float left
-        width 50%
         height onoffswitch-element-height
         padding 0
         line-height onoffswitch-element-height
@@ -36,24 +42,32 @@
         box-sizing border-box
     
     .onoffswitch-inner-on
-        padding-left 5px
+        width s('calc(50% - %s / 2)', onoffswitch-switch-outer-width)
+        padding-left 0.4em
         background-color $action_3
     
     .onoffswitch-inner-off
-        content "OFF"
-        padding-right 5px
+        width s('calc(50% - %s / 2)', onoffswitch-switch-outer-width)
+        padding-right 0.4em
         background-color $balance_8
         text-align right
     
+    .onoffswitch-switch-outer
+        float left
+        width onoffswitch-switch-outer-width
+        background-color: black
+        height onoffswitch-element-height
+        background-image: linear-gradient(to right, $action_3, $action_3 50%, $balance_8 50%, $balance_8 50%)
+
     .onoffswitch-switch
         display block
-        width onoffswitch-switch-width
-        margin 5px
+        margin-left: onoffswitch-switch-unchecked-left-margin
+        margin-top: onoffswitch-switch-inner-margin
+        box-sizing border-box
+        height onoffswitch-switch-inner-height
+        padding 0
         background $balance_11
-        position absolute
-        top 0
-        bottom 0
-        right onoffswitch-switch-right
+        width onoffswitch-switch-width
         border 2px solid $balance_13
         border-radius 0px
-        transition all onoffswitch-transition-duration ease-in 0s
+        transition margin onoffswitch-transition-duration ease-in 0s
\ No newline at end of file
Index: lib/on-off-switch/translations/en_US.coffee
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/on-off-switch/translations/en_US.coffee	(revision dcc2774e5003aecfae162b6da17727cc6b5fb483)
+++ lib/on-off-switch/translations/en_US.coffee	(revision )
@@ -1,3 +1,4 @@
 module.exports =
     onoffswitch_label_on: 'ON'
-    onoffswitch_label_off: 'OFF'
\ No newline at end of file
+    onoffswitch_label_off: 'OFF'
+#    onoffswitch_label_off: 'OFFOFFOFFOFFOFFOFFOFFOFFOFFOFFOFFOFFOFFOFFOFFOFFOFFOFF'
\ No newline at end of file
