Index: pe_rplanng/rplan-jira-client/rplan-jira-client-adapter/src/main/java/de/actano/rplan/jira/adapter/business/UpdateScheduleCommand.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- pe_rplanng/rplan-jira-client/rplan-jira-client-adapter/src/main/java/de/actano/rplan/jira/adapter/business/UpdateScheduleCommand.java	(revision 9566)
+++ pe_rplanng/rplan-jira-client/rplan-jira-client-adapter/src/main/java/de/actano/rplan/jira/adapter/business/UpdateScheduleCommand.java	(revision )
@@ -570,8 +570,28 @@
             if (null != jiraIssueTask)
             {
                 if (null == parentOfJiraIssue) {
-                    // no parent issue -> set the group task as parent
+                    // no parent issue, check if we have an epic link and set it as parent
+                    Object epicLink = taskContext.getOtherFieldFromIssue("Epic Link", jiraIssue);
+                    if (null != epicLink) {
+                        // the epicLink is a jira key, find it in jiraIssues
+                        Task epicLinkTask = null;
+                        for (Issue issue : jiraIssues)
+                        {
+                            if (epicLink.equals(issue.getKey())) {
+                                epicLinkTask = taskContext.getTaskByJiraId(issue.getId().toString());
+                            }
+                        }
+                        if (null != epicLinkTask) {
+                            updateParent(jiraIssueTask, epicLinkTask);
+                        } else {
+                            // epic link not in import -> set the group task as parent
-                    updateParent(jiraIssueTask, taskContext.getIssueGroupTask());
+                            updateParent(jiraIssueTask, taskContext.getIssueGroupTask());
+                        }
+                    } else
+                    {
+                        // no parent issue, no epic link -> set the group task as parent
+                        updateParent(jiraIssueTask, taskContext.getIssueGroupTask());
+                    }
                 } 
                 else 
                 {
\ No newline at end of file
