Index: lib/page-profile/teams/details/invitation/index.coffee
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/page-profile/teams/details/invitation/index.coffee	(revision 59da2ff57c52912a1a31fdc4f9dd9aeaf1bcb3b8)
+++ lib/page-profile/teams/details/invitation/index.coffee	(revision )
@@ -3,6 +3,7 @@
 {connect} = require 'react-redux'
 invitationLinkGenerator = require './external-invitation-link'
 React = require 'react'
+ReactCSSTransitionGroup = require 'react/lib/ReactCSSTransitionGroup'
 r = require 'r-dom'
 
 {CheckBox} = require 'form/form-components'
@@ -40,8 +41,22 @@
         unless @props.memberToken
             linkInputClass += ' disabled'
         r.div
+            key: 'invitation'
             className: 'form invitation'
             [
+                r ReactCSSTransitionGroup,
+                    transitionName: 'invitationTransition'
+                    transitionEnterTimeout: 1000
+                    transitionLeaveTimeout: 1000
+                    transitionAppear: true,
+                    [
+                        r.div
+                            key: 'divKey'
+                            ['bla', 'blub']
+#                        r CheckBox,
+#                            checkedText: 'An'
+#                            unCheckedText: 'Aus'
+                    ]
                 r Row, {className: 'singleline'}, [
                     r.div {className: 'invitationTitle'}, [
                         r CheckBox,
